src/parser/mod.rs

```diff
@@ -6794,7 +6839,23 @@ impl<'a> Parser<'a> {
                         Ok(DataType::Int4(optional_precision?))
                     }
                 }
+                Keyword::INT8 => {
+                    if dialect_of!(self is ClickHouseDialect) {
+                        Ok(DataType::Int8(None))
+                    } else {
+                        let optional_precision = self.parse_optional_precision();
+                        if self.parse_keyword(Keyword::UNSIGNED) {
+                            Ok(DataType::UnsignedInt8(optional_precision?))
+                        } else {
+                            Ok(DataType::Int8(optional_precision?))
+                        }
+                    }
+                }
```

Comment:
```suggestion
                    {
                        let optional_precision = self.parse_optional_precision();
                        if self.parse_keyword(Keyword::UNSIGNED) {
                            Ok(DataType::UnsignedInt8(optional_precision?))
                        } else {
                            Ok(DataType::Int8(optional_precision?))
                        }
                }
```
similar to the string one, we can also have this parse without knowledge of clickhouse?

Reply:
Thanks for a suggestion!
Updated.